src/lib/operators/table_scan.cpp

```diff
@@ -40,13 +46,53 @@ class BaseTableScanImpl {
   virtual std::shared_ptr<const Table> execute() = 0;
 };
 
+// T = datatype
 template <typename T>
 class TableScanImpl : public BaseTableScanImpl {
  public:
+  TableScanImpl<T>(const std::shared_ptr<AbstractOperator> in, const ColumnID column_id, const ScanType scan_type,
+                   const AllTypeVariant search_value)
+      : _in(in), _column_id(column_id), _scan_type(scan_type), _search_value(search_value) {}
+
   std::shared_ptr<const Table> execute() override {
-    // FIXME TODO magic
-    return std::make_shared<Table>();
+    T search_value = type_cast<T>(_search_value);
+    auto pos_list = std::make_shared<PosList>();
+    auto result_table = std::make_shared<Table>();
+    const auto in_table = _in->get_output();
+
+    for (ChunkID chunk_id{0}; in_table->chunk_count(); chunk_id++) {
+      const Chunk& chunk = in_table->get_chunk(chunk_id);
+      auto base_column = chunk.get_column(_column_id);
+      auto value_column = std::dynamic_pointer_cast<ValueColumn<T>>(base_column);
+
+      if (value_column != nullptr) {
+        const std::vector<T>& col_values = value_column->values();
+        for (ChunkOffset chunk_offset; chunk_offset < col_values.size(); chunk_offset++) {
+          auto comparing_function = operators::get_comparator<T>(_scan_type);
+          if (comparing_function(col_values[chunk_offset], search_value)) {
+            pos_list->emplace_back(RowID{chunk_id, chunk_offset});
+          }
+        }
+      }
+
+      auto dictionary_column = std::dynamic_pointer_cast<DictionaryColumn<T>>(base_column);
+      if (dictionary_column != nullptr) {
+        // FIXME TODO magic
+      }
+    }
+
+    auto ref_column = std::make_shared<ReferenceColumn>(in_table, _column_id, pos_list);
```

Comment:
Hier passt es noch nicht zu 100%: der Table-Scan beinhaltet keine Projektion auf die Spalte, welchen den Suchwert beinhalten sollte. Daher müsste man `ReferenceColumn`s in der Anzahl der Spalten der Quelltabelle anlegen und den Spalten-Index als `referenced_column_id` mitgeben. Diese dürfen sich aber ein und dieselbe Positionsliste teilen.